test/framework/controlplane_helpers.go

```diff
@@ -147,6 +147,49 @@ func WaitForOneKubeadmControlPlaneMachineToExist(ctx context.Context, input Wait
 	}, intervals...).Should(BeTrue())
 }
 
+// WaitForOneKubeadmControlPlaneInput is the input for WaitForKubeadmControlPlane.
+type WaitForOneKubeadmControlPlaneInput struct {
+	Lister  Lister
+	Cluster *clusterv1.Cluster
+}
+
+// WaitForOneKubeadmControlPlane will wait until the KCP object is initialized and all control plane machines have node refs.
+func WaitForOneKubeadmControlPlane(ctx context.Context, input WaitForOneKubeadmControlPlaneInput, intervals ...interface{}) *controlplanev1.KubeadmControlPlane {
```

Comment:
WaitForOneKubeadmControlPlane is kind of confusing for me, given that there is always only one KCP for each cluster; also, the func description states that we are waiting for all control plane machines to have node refs, but we are not testing count against KCP.replicas.

Given the problem statement in the PR description, I would suggest a simpler solution described in the previous comment.